### PR TITLE
Feature 1936 madis nc

### LINKED ISSUE #1936: Enhance MADIS2NC to handle the 2016 changes to its format.
## Describe the Problem ##
The issue arose from [METplus Discussion #1189 ](https://github.com/dtcenter/METplus/discussions/1189). The madis2nc tool rejects certain mesonet data from MADIS due to a change in variables/format to the NetCDF files. Sometime in 2016, a change was made to the mesonet output files where they no longer contain variables such as precip3hr, precip6hr, etc. The MADIS support team and confirmed this was an intentional update and said that they do add/change variables to the raw data from time to time, which complicates the current methods for identifying MADIS data type.

The change in data files can be demonstrated in the following files.  On sencea in /d1/projects/METplus/discussions/1189, I ran the following

```
wget https://madis-data.ncep.noaa.gov/madisPublic1/data/archive/2016/01/01/LDAD/mesonet/netCDF/20160101_0000.gz
gunzip 20160101_0000.gz
../../../MET/MET_releases/met-10.1.0-beta2/bin/madis2nc 20160101_0000 20160101_0000.nc -type mesonet

```
and confirmed this produced a successful run.

```
wget https://madis-data.ncep.noaa.gov/madisPublic1/data/archive/2017/01/01/LDAD/mesonet/netCDF/20170101_0000.gz
gunzip 20170101_0000.gz
../../../MET/MET_releases/met-10.1.0-beta2/bin/madis2nc 20170101_0000 20170101_0000.nc -type mesonet

```
and confirmed that this produced the following error and warnings:

```
DEBUG 1: Reading MADIS File:	20170101_0000
ERROR  : 
ERROR  : process_madis_mesonet() Please check if the input is a MESONET.
ERROR  : 
WARNING:     missing variable: precip3hr
WARNING:     missing variable: precip6hr
WARNING:     missing variable: precip12hr
WARNING:     missing variable: precip10min
```

We want to fix main_v10.0 for MET version 10.0.1 and develop for MET version 10.1.0.

### Expected Behavior ###
MADIS2NC should be able to read various mesonet data formats from MADIS.

### Environment ###
Describe your runtime environment:
*1. Machine: (e.g. HPC name, Linux Workstation, Mac Laptop)* Seneca
*2. OS: (e.g. RedHat Linux, MacOS)* Linux
*3. Software version number(s)*

### To Reproduce ###
Describe the steps to reproduce the behavior: **(see above)**
*1. Go to '...'*
*2. Click on '....'*
*3. Scroll down to '....'*
*4. See error*
*Post relevant sample data following these instructions:*
*https://dtcenter.org/community-code/model-evaluation-tools-met/met-help-desk#ftp*

### Relevant Deadlines ###
MET-10.1.0-beta4 (11/15/21)

### Funding Source ###
2702691

### Assignee ###
- [x] Select **engineer(s)** or **no engineer** required
- [ ] Select **scientist(s)** or **no scientist** required

### Labels ###
- [x] Select **component(s)**
- [x] Select **priority**
- [x] Select **requestor(s)**

### Projects and Milestone ###
- [x] Select **Organization** level **Project** for support of the current coordinated release
- [x] Select **Repository** level **Project** for development toward the next official release or add **alert: NEED PROJECT ASSIGNMENT** label
- [x] Select **Milestone** as the next bugfix version

## Define Related Issue(s) ##
Consider the impact to the other METplus components.
- [x] [METplus](https://github.com/dtcenter/METplus/issues/new/choose), [MET](https://github.com/dtcenter/MET/issues/new/choose), [METdatadb](https://github.com/dtcenter/METdatadb/issues/new/choose), [METviewer](https://github.com/dtcenter/METviewer/issues/new/choose), [METexpress](https://github.com/dtcenter/METexpress/issues/new/choose), [METcalcpy](https://github.com/dtcenter/METcalcpy/issues/new/choose), [METplotpy](https://github.com/dtcenter/METplotpy/issues/new/choose)

## Bugfix Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [ ] Complete the issue definition above, including the **Time Estimate** and **Funding Source**.
- [ ] Fork this repository or create a branch of **main_\<Version>**.
Branch name: `bugfix_<Issue Number>_main_<Version>_<Description>`
- [ ] Fix the bug and test your changes.
- [ ] Add/update log messages for easier debugging.
- [ ] Add/update unit tests.
- [ ] Add/update documentation.
- [ ] Push local changes to GitHub.
- [ ] Submit a pull request to merge into **main_\<Version>**.
Pull request: `bugfix <Issue Number> main_<Version> <Description>`
- [ ] Define the pull request metadata, as permissions allow.
Select: **Reviewer(s)** and **Linked issues**
Select: **Organization** level software support **Project** for the current coordinated release
Select: **Milestone** as the next bugfix version
- [ ] Iterate until the reviewer(s) accept and merge your changes.
- [ ] Delete your fork or branch.
- [ ] Complete the steps above to fix the bug on the **develop** branch.
Branch name:  `bugfix_<Issue Number>_develop_<Description>`
Pull request: `bugfix <Issue Number> develop <Description>`
Select: **Reviewer(s)** and **Linked issues**
Select: **Repository** level development cycle **Project** for the next official release
Select: **Milestone** as the next official version
- [ ] Close this issue.

### PR DESCRIPTION
**Pull request by hsoh-u**
## Pull Request Testing ##

- [x] Describe testing already performed for these changes:</br>

  - Tested with 2017 & 2021 mesonet data
  - set MET_BASE for the updated default configuration file

export MET_BASE=/d1/personal/hsoh/git/features/feature_1936_madis_nc/MET/met/share/met
/d1/personal/hsoh/git/features/feature_1936_madis_nc/MET/met/bin/madis2nc /d1/projects/MET/MET_test_data/unit_test/obs_data/madis/mesonet/mesonet_20170101_0000.nc out_mesonet_20170101_0000.nc -type mesonet

/d1/personal/hsoh/git/features/feature_1936_madis_nc/MET/met/bin/madis2nc /d1/personal/hsoh/data/MET-1936/20210101_0000 out_mesonet_20210101_0000.nc -type mesonet

unit test : unit_madis2nc.xml

- [x] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>

Review document at 6.3.2. madis2nc configuration file.

- [x] Do these changes include sufficient documentation updates, ensuring that no errors or warnings exist in the build of the documentation? **[Yes]**

Added mesonet_opt_var  at 6.3.2. madis2nc configuration file.

- [x] Do these changes include sufficient testing updates? **[Yes]**

A unit test, madis2nc_MESONET_optional_vars, is added

- [x] Will this PR result in changes to the test suite? **[Yes]**</br>
If **yes**, describe the new output and/or changes to the existing output:</br>

One more output: mesonet_20170101_0000_F000.nc

- [ ] Please complete this pull request review by **[Fill in date]**.</br>

## Pull Request Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [ ] Complete the PR definition above.
- [ ] Ensure the PR title matches the feature or bugfix branch name.
- [ ] Define the PR metadata, as permissions allow.
Select: **Reviewer(s)**
Select: **Organization** level software support **Project** or **Repository** level development cycle **Project**
Select: **Milestone** as the version that will include these changes
- [ ] After submitting the PR, select **Linked issues** with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue and delete your feature or bugfix branch from GitHub.

### CODE REVIEW

File: met/docs/Users_Guide/reformat_point.rst [hunk 1]
```diff
@@ -704,6 +704,16 @@ _________________
 
 The configuration options listed above are common to many MET tools and are described in :numref:`config_options`.
 
+_________________
+
+
+.. code-block:: none
+
+		mesonet_opt_var = [ "precip3hr", "precip6hr", "precip12hr", "precip10min" ];
```
Comment (JohnHalleyGotway) [inline met/docs/Users_Guide/reformat_point.rst:712]:
```suggestion
    mesonet_opt_var = [ "precip3hr", "precip6hr", "precip12hr", "precip10min" ];
```

**Comment by JohnHalleyGotway** [comment]
@hsoh-u so what are we going to do about this PR? Is there a way of supporting multiple versions of these input files without needing to add a new config file option? It'd be preferable to only add new config options when really needed. Hopefully there's a solution that doesn't require users to be aware of the specific details of the contents of the input files.

If needed, we could shift this issue and PR to the beta5 development cycle instead.

**Comment by JohnHalleyGotway** [comment]
@hsoh-u is this pull request superceded by dtcenter/MET#1984?

If so, I recommend that you close it without merging.

**Comment by hsoh-u** [comment]
The default configuration will not be updated for this. Superseded by #1984

**Pull request closed without merging**